### ISSUE TITLE
Cozy-indexer config file not removed by purge

### ISSUE THREAD

**Issue by nicofrand** [cozy/cozy-debian#15]
It seems that the purge command does not properly remove the cozy-indexer config files (same for cozy-controller).

Here are the logs from `apt-get purge cozy` instantly followed by `apt-get install cozy` : 

``` console
# apt-get purge couchdb cozy
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package 'couchdb' is not installed, so not removed
The following packages were automatically installed and are no longer required:
  coffeescript couchdb-bin couchdb-common cozy-indexer gir1.2-glib-2.0 gyp libc-ares-dev libc-ares2 libdbus-glib-1-2
  libgirepository-1.0-1 libjs-node-uuid libjs-underscore libmysqlclient18 libv8-3.14-dev libv8-3.14.5 mysql-common node-abbrev
  node-ansi node-ansi-color-table node-archy node-async node-block-stream node-combined-stream node-cookie-jar
  node-delayed-stream node-forever-agent node-form-data node-fstream node-fstream-ignore node-github-url-from-git node-glob
  node-graceful-fs node-gyp node-inherits node-ini node-json-stringify-safe node-lockfile node-lru-cache node-mime
  node-minimatch node-mkdirp node-mute-stream node-node-uuid node-nopt node-normalize-package-data node-npmlog node-once
  node-osenv node-qs node-read node-read-package-json node-request node-retry node-rimraf node-semver node-sha node-sigmund
  node-slide node-tar node-tunnel-agent node-underscore node-which pwgen python-cffi python-chardet python-chardet-whl
  python-colorama python-colorama-whl python-cryptography python-dateutil python-distlib python-distlib-whl python-html5lib
  python-html5lib-whl python-meld3 python-mysqldb python-ndg-httpsclient python-openssl python-pip python-pip-whl
  python-pkg-resources python-ply python-pyasn1 python-pycparser python-pycurl python-requests python-requests-whl
  python-setuptools python-setuptools-whl python-six python-six-whl python-software-properties python-support python-tornado
  python-urllib3 python-urllib3-whl python-virtualenv python-wheel python-whoosh python3-apt python3-dbus python3-gi
  python3-pkg-resources python3-software-properties python3-virtualenv software-properties-common sqlite3 sudo supervisor
  unattended-upgrades virtualenv
Use 'apt-get autoremove' to remove them.
The following packages will be REMOVED:
  cozy
0 upgraded, 0 newly installed, 1 to remove and 0 not upgraded.
1 not fully installed or removed.
After this operation, 48.1 kB disk space will be freed.
Do you want to continue? [Y/n] y
(Reading database ... 57342 files and directories currently installed.)
Removing cozy (1.13) ...
npm WARN uninstall not installed in /usr/lib/node_modules: "cozy-controller"
npm WARN uninstall not installed in /usr/lib/node_modules: "cozy-monitor"
Remove user cozy
Remove user cozy-data-system
Remove user cozy-home
Remove /etc/supervisor/conf.d/cozy-indexer.conf
Remove /etc/supervisor/conf.d/cozy-controller.conf
Reload supervisord
Restarted supervisord
Remove /etc/apache2/sites-enabled/cozy.conf and restart Apache
Syntax OK
Remove /etc/apache2/sites-available/cozy.conf

# apt-get install cozy
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  gyp libc-ares-dev libc-ares2 libjs-node-uuid libjs-underscore libv8-3.14-dev libv8-3.14.5 node-abbrev node-ansi
  node-ansi-color-table node-archy node-async node-block-stream node-combined-stream node-cookie-jar node-delayed-stream
  node-forever-agent node-form-data node-fstream node-fstream-ignore node-github-url-from-git node-glob node-graceful-fs
  node-gyp node-inherits node-ini node-json-stringify-safe node-lockfile node-lru-cache node-mime node-minimatch node-mkdirp
  node-mute-stream node-node-uuid node-nopt node-normalize-package-data node-npmlog node-once node-osenv node-qs node-read
  node-read-package-json node-request node-retry node-rimraf node-semver node-sha node-sigmund node-slide node-tar
  node-tunnel-agent node-underscore node-which
Use 'apt-get autoremove' to remove them.
The following extra packages will be installed:
  couchdb
The following NEW packages will be installed:
  couchdb cozy
0 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/15.9 kB of archives.
After this operation, 142 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
Preconfiguring packages ...
Selecting previously unselected package couchdb.
(Reading database ... 57343 files and directories currently installed.)
Preparing to unpack .../couchdb_1.6.1-2_all.deb ...
Unpacking couchdb (1.6.1-2) ...
Setting up couchdb (1.6.1-2) ...
Selecting previously unselected package cozy.
(Reading database ... 57353 files and directories currently installed.)
Preparing to unpack .../apt/archives/cozy_1.13_all.deb ...
Unpacking cozy (1.13) ...
Setting up cozy (1.13) ...
Chosen domain name for Cozy: cozy.mondomaine.com
Installing NPM dependencies...
npm WARN engine hawk@0.10.2: wanted: {"node":"0.8.x"} (current: {"node":"0.10.40","npm":"1.4.28"})
npm WARN engine boom@0.3.8: wanted: {"node":"0.8.x"} (current: {"node":"0.10.40","npm":"1.4.28"})
npm WARN engine sntp@0.1.4: wanted: {"node":"0.8.x"} (current: {"node":"0.10.40","npm":"1.4.28"})
npm WARN engine hoek@0.7.6: wanted: {"node":"0.8.x"} (current: {"node":"0.10.40","npm":"1.4.28"})
npm WARN engine cryptiles@0.1.3: wanted: {"node":"0.8.x"} (current: {"node":"0.10.40","npm":"1.4.28"})
/usr/bin/cozy-controller -> /usr/lib/node_modules/cozy-controller/bin/cozy-controller
cozy-controller@2.0.29 /usr/lib/node_modules/cozy-controller
├── rimraf@2.2.8
├── printit@0.1.6
├── mozilla-version-comparator@1.0.2
├── async@0.9.0
├── mkdirp@0.5.0 (minimist@0.0.8)
├── request-json-light@0.5.15 (mime@1.3.4, form-data@0.1.4)
├── cozy-controller-carapace@0.8.0 (eventemitter2@0.4.14, semver@1.0.14, async@0.1.22, optimist@0.3.7)
├── tar@0.1.17 (inherits@1.0.0, block-stream@0.0.8, fstream@0.1.31)
├── americano@0.3.10 (errorhandler@1.0.1, commander@2.2.0, printit@0.1.3, method-override@2.0.0, morgan@1.1.1, body-parser@1.3.0, express@4.4.0)
├── request@2.53.0 (caseless@0.9.0, aws-sign2@0.5.0, forever-agent@0.5.2, form-data@0.2.0, stringstream@0.0.4, oauth-sign@0.6.0, tunnel-agent@0.4.1, isstream@0.1.2, json-stringify-safe@5.0.1, node-uuid@1.4.3, qs@2.3.3, combined-stream@0.0.7, mime-types@2.0.14, http-signature@0.10.1, bl@0.9.4, tough-cookie@2.0.0, hawk@2.3.1)
├── americano-cozy@0.2.8 (printit@0.1.3, jugglingdb@0.2.14, jugglingdb-cozy-adapter@0.4.12)
├── forever-monitor@1.3.0 (watch@0.5.1, pkginfo@0.3.0, minimatch@0.2.14, utile@0.1.7, ps-tree@0.0.3, broadway@0.2.10)
└── npm@1.4.28
/usr/bin/cozy-monitor -> /usr/lib/node_modules/cozy-monitor/bin/cozy-monitor
cozy-monitor@1.2.27 /usr/lib/node_modules/cozy-monitor
├── printit@0.1.3
├── colors@1.1.2
├── async@0.9.0
├── always-tail@0.1.1 (debug@0.7.4)
├── commander@1.0.5 (keypress@0.1.0)
├── shelljs@0.3.0
├── request-json-light@0.5.10 (mime@1.3.4, form-data@0.1.4)
├── coffee-script@1.9.0
├── axon@0.6.1 (escape-regexp@0.0.1, configurable@0.0.1, debug@2.2.0)
└── cozy-clients@1.0.3 (request-json@0.4.10)
Creating UNIX users...
Generating CouchDB tokens...
Configuring Cozy Indexer...
Not replacing deleted config file /etc/supervisor/conf.d/cozy-indexer.conf
sha1sum: /etc/supervisor/conf.d/cozy-indexer.conf: No such file or directory
dpkg: error processing package cozy (--configure):
 subprocess installed post-installation script returned error exit status 1
Errors were encountered while processing:
 cozy
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

**Comment by nledez** [comment]
May be fixed in ab8cadc01f9002dcb5b793aa7312874fe71a6259.

**Comment by clochix** [comment]
Closing, as the Node.js version of Cozy is now deprecated.
